### PR TITLE
Added clearer error messages for the nRequest amount validation

### PR DESCRIPTION
**Pull request by mxaddict**

### CODE REVIEW

File: src/qt/communityfundcreatepaymentrequestdialog.cpp [hunk 1]
```diff
@@ -192,8 +194,11 @@ void CommunityFundCreatePaymentRequestDialog::click_pushButtonSubmitPaymentReque
         // Validate requested amount
         if (nReqAmount <= 0 || nReqAmount > proposal.GetAvailable(*pcoinsTip, true)) {
             QMessageBox msgBox(this);
-            std::string str = "Cannot create a Payment Request for the requested amount\n";
-            msgBox.setText(tr(str.c_str()));
+            std::string str = "Requested amount must be greater than 0 NAV (Zero)\n";
+            if (nReqAmount > proposal.GetAvailable(*pcoinsTip, true)) {
+                str = "Requested amount %1 is more than avaible coins in the proposal (%2)\n";
+            }
+            msgBox.setText(tr(str.c_str()).arg(NavCoinUnits::formatWithUnit(NavCoinUnits::NAV, nReqAmount), NavCoinUnits::formatWithUnit(NavCoinUnits::NAV, proposal.GetAvailable(*pcoinsTip, true))));
```
[the remark this answers is not in the document]
Comment (mxaddict) [reply src/qt/communityfundcreatepaymentrequestdialog.cpp]: @aguycalled you mean the `tr()` call needs to have the string passed inline? and not via variable?

[the remark this answers is not in the document]
Comment (aguycalled) [reply src/qt/communityfundcreatepaymentrequestdialog.cpp]: yep iirc

[the remark this answers is not in the document]
Comment (mxaddict) [reply src/qt/communityfundcreatepaymentrequestdialog.cpp]: Like this:
```cpp
            QString str = tr("Requested amount must be greater than 0 NAV (Zero)\n");
            if (nReqAmount > proposal.GetAvailable(*pcoinsTip, true)) {
                str = tr("Requested amount %1 is more than avaible coins in the proposal (%2)\n")
                    .arg(
                        NavCoinUnits::formatWithUnit(NavCoinUnits::NAV, nReqAmount),
                        NavCoinUnits::formatWithUnit(NavCoinUnits::NAV, proposal.GetAvailable(*pcoinsTip, true))
                    );
            }
            msgBox.setText(str);
```

**Comment by mxaddict** [comment]
Fixed merge conflict

**Comment by mxaddict** [comment]
Lets merge this after 4.7.1 release

**Comment by dantudor** [comment]
str = "Requested amount %1 is more than avaible coins in the proposal (%2)\n";
Should be:
str = "Requested amount %1 is more than availble coins in the proposal (%2)\n";

**Comment by chasingkirkjufell** [comment]
good catch but isn't it actually "available" ?

**Pull request merged**